### ISSUE TITLE
Prepare the rules for ESLint v9

### ISSUE THREAD

**Issue by Belco90** [testing-library/eslint-plugin-testing-library#899]
### Plugin version

n/a

### What problem do you want to solve?

- [Preparing your custom rules for ESLint v9.0.0](https://eslint.org/blog/2023/09/preparing-custom-rules-eslint-v9/)

### Your take on the correct solution?

Apply the changes mentioned in the linked migration docs so the custom rules are adapted to ESLint v9 and the

### Do you want to submit a pull request to implement this change?

Yes

**Comment by thenbe** [comment]
An example of a similar migration on another eslint plugin: https://github.com/eslint-community/eslint-plugin-security/pull/145/files

**Comment by Belco90** [comment]
@thenbe Thanks for the example! It will definitely help when migrating this plugin.

**Comment by thenbe** [comment]
For anyone tracking this PR: eslint released some [compatibility utilities](https://eslint.org/blog/2024/05/eslint-compatibility-utilities/) that allows users to use this plugin in its current state with eslint v9.

tldr:

```diff
// eslint.config.js
+ import { fixupPluginRules } from '@eslint/compat';

	{
		files: ['**/*.test.{js,ts,jsx,tsx}'],
		plugins: {
-			'testing-library': {
-				rules: testingLibrary.rules,
-			},
+			'testing-library': fixupPluginRules({
+				rules: testingLibrary.rules,
+			}),
		},
		rules: testingLibrary.configs.react.rules,
	}
```

**Comment by jagretz** [comment]
> For anyone tracking this PR: eslint released some [compatibility utilities](https://eslint.org/blog/2024/05/eslint-compatibility-utilities/) that allows users to use this plugin in its current state with eslint v9.

Thank you for this. I didn't remember this existed when I went through the flat-config docs. 

For anyone using with `eslint-plugin-import`, when I add the compat lib, I receive an eslint error above the import statement:

```js 
// Parse errors in imported module '@eslint/compat': parserPath or languageOptions.parser is required! (undefined:undefined)  import/namespace
import { fixupPluginRules } from "@eslint/compat";
```

It's important to note that `eslint-plugin-import` is (also) not quite ready for flat-config, however, I tested a few rules from `eslint-plugin-testing-library` and the solution offered by @thenbe is working properly based on my limited testing.

**Comment by SalahAdDin** [comment]
is there Any ETA for it?

**Comment by G-Rath** [comment]
I have opened #925 which should resolve this without requiring a new major

**Comment by dosolkowski-work** [comment]
I know this was technically closed by PR #928, but due to outdated dependencies there are still errors attempting to install the latest version with ESLint v9:

```
 WARN  Issues with peer dependencies found
.
└─┬ eslint-plugin-testing-library 6.4.0
  └─┬ @typescript-eslint/utils 5.62.0
    └── ✕ unmet peer eslint@"^6.0.0 || ^7.0.0 || ^8.0.0": found 9.13.0
```

Maybe this depends on issue #855 before ESLint v9 is fully supported?

**Comment by MichaelDeBoey** [comment]
@dosolkowski-work Technically, ESLint v9 is now supported as all rules are updated

Having a (peer)dependency nog support it (officially), doesn't make that less true.

We are however going to release a major version where all dependencies are updated to their latest version (together with some other breaking changes), which will at that point not throw the warning anymore

It's a warning, so you can just ignore it (for now) 😉
